File: Ahorn/entities/torch.jl
module JungleHelperTorch

using ..Ahorn, Maple

@mapdef Entity "JungleHelper/Torch" Torch(x::Integer, y::Integer, flag::String="torch_flag", sprite::String="")

const placements = Ahorn.PlacementDict(
    "Torch (Jungle Helper)" => Ahorn.EntityPlacement(
        Torch
    )
)

function Ahorn.selection(entity::Torch)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 9, y - 22, 17, 22)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Torch, room::Maple.Room)
    Ahorn.drawSprite(ctx, "JungleHelper/TorchNight/TorchNightOff", 0, -11)
end

end

File: Loenn/entities/torch.lua
local utils = require("utils")

local torch = {}
torch.name = "JungleHelper/Torch"
torch.placements = {
    {
        name = "default",
        data = {
            flag = "torch_flag",
            sprite = "",
        }
    }
}

torch.offset = { 16, 27 }
torch.texture = "JungleHelper/TorchNight/TorchNightOff"

function torch.selection(room, entity)
    return utils.rectangle(entity.x - 9, entity.y - 22, 17, 22)
end

return torch
